airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -354,18 +355,23 @@ def trigger_dag_run(
             f"DAG with dag_id: '{dag_id}' has import errors and cannot be triggered",
         )
 
-    logical_date = pendulum.instance(body.logical_date)
-
+    logical_date = pendulum.instance(body.logical_date) if body.logical_date else None
+    run_after = timezone.coerce_datetime(timezone.utcnow())
```

Comment:
This should use run_after from the post body.

Reply:
Changed in https://github.com/apache/airflow/pull/46616